config/src/lib.rs

```diff
@@ -146,6 +146,14 @@ pub struct BlockSynchronizerParameters {
     /// discover who has the payload available for the dictated certificates.
     #[serde(with = "duration_format")]
     pub payload_availability_timeout: Duration,
+    /// When a certificate is fetched on the fly from peers, it is submitted
+    /// from the block synchronizer handler for further processing to core
+    /// to validate and ensure parents are available and history is causal
+    /// complete. This property is the timeout while we wait for core to
+    /// perform this processes and the certificate to become available to
+    /// the handler to consume.
+    #[serde(with = "duration_format")]
+    pub handler_certificate_deliver_timeout: Duration,
```

Comment:
I've got no problem doing this. My only "objection" is that even to explain the whole thing I had/need to get into details (after all the properties are under the `BlockSynchronizerParameters` which is an implementation detail) so don't believe it makes that much of a difference. Having said that, another +1 on your point about having a unified timeout parameter instead of multiple ones here and there